### PR TITLE
logging improvements

### PR DESCRIPTION
**Pull request by k9ert**
* LogLevel changeable at runtime in settings-menu
* A lot of replacements of print statements --> logging
* Specifying a default-logging consistently for app.logger + logging.root_logger

Hopefully we'll have some more reasonable output in case of issues. Still far from perfect, though.

It's not yet defining the start-loglevel-state properly. Currently it seems to be "DEBUG" for an unknown reason. I'll have a look at that and please consider this as WIP in the meantime.

### CODE REVIEW

**Comment by k9ert** [comment]
Created explicit tests in the settings-menu after you click on the test-button:
image

**Comment by stepansnigirev** [comment]
Looks good.
I just merged #136 so now it requires rebase :(
After rebase I think we can merge it.

**Comment by ben-kaufman** [comment]
One thing I noticed is that the checkboxes are clickable, and will take you to an invalid page URL, I guess that should be disabled. Besides that looks great.

**Comment by k9ert** [comment]
Uaahh, my fault. Should have avoided the merge commit. Stay tuned while rebasing ...

File: src/cryptoadvance/specter/controller.py [hunk 1]
```diff
@@ -385,20 +397,22 @@ def wallet(wallet_alias):
...
-    if wallet is None:
-        return render_template("base.jinja", error="Wallet not found", specter=app.specter, rand=rand)
-
-    return render_template("wallet/history/txs/wallet_tx.jinja", wallet_alias=wallet_alias, wallet=wallet, specter=app.specter, rand=rand)
+    try:
+        wallet = app.specter.wallets.get_by_alias(wallet_alias)
+    except SpecterError as se:
+        app.logger.error("SpecterError while wallet_tx: %s" % se)
+        return render_template("base.jinja", error=se, specter=app.specter, rand=rand)
+    return render_template("wallet_tx.html", wallet_alias=wallet_alias, wallet=wallet, specter=app.specter, rand=rand)
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/controller.py:405]: Should be 
```py
   return render_template("wallet/history/txs/wallet_tx.jinja", wallet_alias=wallet_alias, wallet=wallet, specter=app.specter, rand=rand)
 ```

File: src/cryptoadvance/specter/controller.py [hunk 2]
```diff
@@ -36,15 +34,18 @@
...
 
 @app.route('/wallets/<wallet_alias>/combine/', methods=['GET', 'POST'])
 @login_required
 def combine(wallet_alias):
-    wallet = app.specter.wallets.get_by_alias(wallet_alias)
+    try:
+        wallet = app.specter.wallets.get_by_alias(wallet_alias)
+    except SpecterError as se:
+        app.logger.error("SpecterError while combine: %s" % se)
+        return render_template("base.html", error=se, specter=app.specter, rand=rand)
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/controller.py:48]: should be `base.jinja`

File: src/cryptoadvance/specter/controller.py [hunk 3]
```diff
@@ -65,7 +66,11 @@ def combine(wallet_alias):
 @app.route('/wallets/<wallet_alias>/broadcast/', methods=['GET', 'POST'])
 @login_required
 def broadcast(wallet_alias):
-    wallet = app.specter.wallets.get_by_alias(wallet_alias)
+    try:
+        wallet = app.specter.wallets.get_by_alias(wallet_alias)
+    except SpecterError as se:
+        app.logger.error("SpecterError while broadcast: %s" % se)
+        return render_template("base.html", error=se, specter=app.specter, rand=rand)
```
Comment (stepansnigirev) [inline src/cryptoadvance/specter/controller.py:73]: should be `base.jinja`

**Pull request merged**